utils/streams.js

```diff
@@ -0,0 +1,212 @@
+const program = require("commander");
+const through = require("through2");
+const fs = require("fs");
+const path = require("path");
+
+const Importer = require("../modules/importer").default;
+const myImporter = new Importer({});
+
+const actions = [
+	{
+		name: "reverse",
+		run: () => {
+			process.stdin.setEncoding("utf8");
+
+			process.stdin.on("readable", () => {
+				var chunk = process.stdin.read();
+				if (chunk !== null) {
+					const str = chunk.toString().trim();
+
+					if (str === "exit") {
+						process.stdin.end(null);
+						process.exit();
+					}
+
+					let result = "";
+					let i = str.length;
+
+					while (i--) {
+						result += str[i];
+					}
+
+					process.stdout.write(`Result: ${result}\n\n`);
+				}
+			});
+		}
+	},
+	{
+		name: "transform",
+		run: () => {
+			const processing = through(write, end);
+
+			process.stdin.pipe(processing).pipe(process.stdout);
+
+			function write(buffer, encoding, next) {
+				const str = buffer.toString().trim();
+
+				if (!str || str === "exit") {
+					this.push(null);
+					process.exit();
+				}
+
+				next(null, `Result: ${str.toUpperCase()}\n\n`);
+			}
+
+			function end(done) {
+				done();
+			}
+		}
+	},
+	{
+		name: "outputFile",
+		run: options => {
+			if (!options.file) {
```

Comment:
As I see, this code is used at least in 3 places. Please, extract it to separate function